Sources/SwiftFusion/Geometry/Pose2.swift

```diff
@@ -114,106 +192,78 @@ public struct Pose2: Differentiable, TangentStandardBasis, KeyPathIterable {
-  /// translation.
-  ///
-  /// See the documentation comment on the declaration of `Pose2` for more information.
-  public typealias TangentVector = Vector3
-
-  /// Creates a tangent vector given rotation (`w`) and translation (`v`) components.
-  public static func tangentVector(w: Vector1, v: Vector2) -> Vector3 {
-    Vector3(w.x, v.x, v.y)
+// MARK: Coordinate Operators
+public extension Pose2Coordinate {
+  /// Product of two rotations.
+  @differentiable
+  static func * (lhs: Pose2Coordinate, rhs: Pose2Coordinate) -> Pose2Coordinate {
+    Pose2Coordinate(lhs.rot * rhs.rot, lhs.t + lhs.rot * rhs.t)
   }
 
-  /// Returns the rotation (`w`) and translation (`v`) components of `tangentVector`.
-  public static func decomposed(tangentVector: Vector3) -> (w: Vector1, v: Vector2) {
-    (Vector1(tangentVector.x), Vector2(tangentVector.y, tangentVector.z))
+  /// Inverse of the rotation.
+  @differentiable
+  func inverse() -> Pose2Coordinate {
+    Pose2Coordinate(self.rot.inverse(), self.rot.unrotate(-self.t))
   }
+}
 
-  /// Moves `self` by `exp(hat(direction))`.
+extension Pose2Coordinate: ManifoldCoordinate {
   public mutating func move(along direction: Vector3) {
-    // TODO: This should be the real exponential map.
     let (w, v) = Self.decomposed(tangentVector: direction)
     self.tStorage.move(along: rot.rotate(v))
     self.rotStorage.move(along: w)
   }
-
-  /// Storage for the pose's translation.
+  
+  /// p * Exp(q)
+  @differentiable(wrt: local)
+  public func global(_ local: Vector3) -> Self {
+    self * Pose2Coordinate(Rot2(local.x), Vector2(local.y, local.z))
+  }
+  
+  /// Log(p^{-1} * q)
   ///
-  /// This is private so that we can define an accessor with a custom derivative (Swift AD does not
-  /// currently support custom derivatives on stored properties).
-  private var tStorage: Vector2
-
-  /// Storage for the pose's rotation.
+  /// Explanation
+  /// ====================
+  /// `global_p(local_p(q)) = q`
+  /// e.g. `p*Exp(Log(p^{-1} * q)) = q`
+  /// QED
   ///
-  /// This is private so that we can define an accessor with a custom derivative (Swift AD does not
-  /// currently support custom derivatives on stored properties).
-  private var rotStorage: Rot2
-}
-
-/// Convenience initializers.
-extension Pose2 {
-  /// Creates a `Pose2` with translation `x` and `y` and with rotation `theta`.
-  @differentiable
-  public init(_ x: Double, _ y: Double, _ theta: Double) {
-    self.init(Rot2(theta), Vector2(x, y))
-  }
-
-  public init(randomWithCovariance covariance: Tensor<Double>) {
-    self.init(0, 0, 0)
-    let r = matmul(cholesky(covariance), Tensor<Double>(randomNormal: [3, 1])).scalars
-    let tv = Pose2.TangentVector(r[0], r[1], r[2])
-    self.move(along: tv)
+  @differentiable(wrt: global)
+  public func local(_ global: Self) -> Vector3 {
+    let d = self.inverse() * global
+    return Vector3(d.rot.theta, d.t.x, d.t.y)
```

Comment:
Of course you'll have to change the log here to be the appropriate inverse if you change the exp above.